### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by zeff96**
- [ ] Your models should include some additional methods.

- Users

     A method that returns the 3 most recent posts for a given user.

- Posts

     A method that updates the posts counter for a user.
     A method which returns the 5 most recent comments for a given post.

- Comments

    A method that updates the comments counter for a post.

- Likes

    A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_one :posts
+  has_many :comments, through: :posts
+  has_many :likes, through: :posts
+
```
Comment (emmanuellekamwa) [inline app/models/user.rb:5]: - [ ] According to the ERD diagram, the relationship between user and post is a `one-to-many relationship`. So change `has_one :posts` to `has_many :posts` and don't forget to add the foreign key `author_id` to all the relationships.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
```
Comment (emmanuellekamwa) [inline app/models/comment.rb:4]: - [ ] Apply the same approach as the post model.

File: app/models/like.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
```
Comment (emmanuellekamwa) [inline app/models/like.rb:4]: - [ ] Apply the same approach here.

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED ❌⛔**
Hello @zeff96 👋
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

## Required Changes ♻️
_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @zeff96, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**